### PR TITLE
Moved positional argument size of sprint() function to sizehint kw

### LINKED ISSUE #24868: sprint: rename `env` keyword, add `size` keyword
As observed in https://github.com/JuliaLang/julia/pull/24727, the `sprint` function takes an `env` keyword which could have a better name, e.g. `context`. All methods should accept that keyword. It would also be better to have a `sizehint` keyword than the awkward leading positional size argument.

### PR DESCRIPTION
**Pull request by evetion**
Also renamed env to context.

Fixes #24868

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

**Comment by StefanKarpinski** [comment]
I will merge this in 24 hours unless there are objections in the mean time!

**Comment by iblis17** [comment]
should this be changed as well?
https://github.com/JuliaLang/julia/blob/622ab915e080b10862f4548f670467b18db55cdd/stdlib/Test/src/Test.jl#L230-L233

File: base/show.jl
Comment (omus) [inline base/show.jl]: I noticed this in my `sprint` PR. Are there any performance gains with only calling `sprint` once here?

Comment (StefanKarpinski) [reply base/show.jl]: Probably but not really relevant to this PR.

**Comment by StefanKarpinski** [comment]
@iblis17: it could be but I don't think it has to be. It's not using a deprecated signature, it's just not using the new signature that does that more conveniently, IIUC.

**Review by omus — APPROVED** [review]

**Comment by StefanKarpinski** [comment]
Sorry this is taking so long, @evetion – there keep being merge conflict and then I run CI again just to be safe. I will do my best to merge this at some point today!

**Comment by evetion** [comment]
No problem @StefanKarpinski, running the CI is good practice. I wonder if it's possible for git to be smarter in detecting two seperate new additions into one file, `base/deprecated.jl` often has merge conflicts. I can imagine other PRs have the same problem.

**Comment by StefanKarpinski** [comment]
There, hopefully I got that merge right.

**Comment by StefanKarpinski** [comment]
* Travis Linux runs both passed

**Comment by StefanKarpinski** [comment]
I'm going to make the judgement call that since this passes on both linuxes and on my local macOS, it's safe to merge. There's not much in here that could possibly be platform-specific.

File: base/arrayshow.jl
Comment (stevengj) [inline base/arrayshow.jl]: Since `sizehint=0` is the default, the keyword could have been omitted here...  presumably, it was only being passed before because that was previously the only way to pass `env`.

**Pull request merged**
